From 0c9717ab949bb57757b2e6f7a5538b615bb4ca30 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Mon, 25 Feb 2013 11:52:49 +0100
Subject: [PATCH] Rework
 ViewHost_PrepareAddWindow/ViewHost_PrepareAddWindowWithMetaData

Both IPC methods are now returning the key of the new window directly instead of letting
the client assign one.

Open-WebOS-DCO-1.0-Signed-Off-By: Simon Busch <morphis@gravedo.de>
---
 include/public/messages/SysMgrMessagesInternal.h |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/public/messages/SysMgrMessagesInternal.h b/include/public/messages/SysMgrMessagesInternal.h
index f5ce3d0..426b143 100644
--- a/include/public/messages/SysMgrMessagesInternal.h
+++ b/include/public/messages/SysMgrMessagesInternal.h
@@ -215,12 +215,12 @@ IPC_BEGIN_MESSAGES(ViewHost)
     IPC_MESSAGE_CONTROL1(View_Host_ReturnedKeyEvent, SysMgrKeyEvent)
 
     // creates a new window (used to create PDK cards, type seems to be irrelevant)
-    // params: window key (shmget), type, width, height
-    IPC_MESSAGE_CONTROL4(ViewHost_PrepareAddWindow, int, int, int, int)
+    // params: type, width, height, window key (shmget)
+    IPC_SYNC_MESSAGE_CONTROL3_1(ViewHost_PrepareAddWindow, int, int, int, int)
 
     // creates a new window (used by WAM)
-    // params: window key (shmget), metaDataKey, type, width, height
-    IPC_MESSAGE_CONTROL5(ViewHost_PrepareAddWindowWithMetaData, int, int, int, int, int)
+    // params:  metaDataKey, type, width, height, window key (shmget)
+    IPC_SYNC_MESSAGE_CONTROL4_1(ViewHost_PrepareAddWindowWithMetaData, int, int, int, int, int)
 
     // the window is ready
     // params: window key
-- 
1.7.10.4

